File: day006/src/main/java/com/pawelpluta/day006/Person.java
package com.pawelpluta.day006;

import org.springframework.data.annotation.Id;

class Person {
    @Id
    private String id;
    private String firstName;
    private String lastName;

    Person(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    String getFirstName() {
        return firstName;
    }

    String getLastName() {
        return lastName;
    }
}

File: day012/src/main/java/com/pawelpluta/day012/Person.kt
package com.pawelpluta.day012

internal data class Person(
        val firstName: String,
        val lastName: String,
        val birthYear: Int,
        val homeAddress: Address,
        val preferredAddress: Address?,
        val siblings: List<Person>) {
    companion object {}

    fun displayName(): String {
        return "$firstName $lastName"
    }

    fun mailingAddress(): Address {
        return preferredAddress ?: homeAddress
    }

    fun olderSiblings(): List<Person> {
        return siblings.filter { it.birthYear < birthYear }
    }
}